pkg/metrics/util_test.go

```diff
@@ -291,3 +293,57 @@ agones_gameservers_node_count_count 3
 agones_nodes_count{empty="false"} 2
 agones_nodes_count{empty="true"} 1
 `
+
+func Test_parseLabels(t *testing.T) {
+	tests := []struct {
```

Comment:
Another way you could write this (to more easily explore edge cases that should create errors) is to have two sets of tests, one that should produce errors and another that should work. I think the set of tests that should work is relatively small (you have 3 right now, which mostly covers it). But the set of inputs that should cause an error should be made much longer. This is easier if it's more compact, e.g.

```
errorInputs = []string{
  "a=b,",
  "a=b,c",
  ...
}
```

since the want is always nil and the wantErr is always true for these you can collapse them into a shorter list form, and then expand the list.

What about testing inputs with spaces added in various places? Or extra equals signs? Should non-ascii characters be accepted? Does stack driver impose any constraints on what characters can be in a key? And so on.